service/src/main/java/gov/nasa/pds/api/registry/business/LidVidUtils.java

```diff
@@ -59,115 +45,92 @@ public static String extractLidFromLidVid(String identifier)
 
+            if (lidvids.isEmpty()) throw new LidVidNotFoundException(lid);
+    		else return lidvids.get(lidvids.size() - 1);
+    	}
+    	throw new LidVidNotFoundException(lid);
+    }
     
-    /**
-     * Build aggregation query to select latest versions of lids
-     * @param lids list of LIDs
-     * @return opensearch query
-     */
-    public static SearchSourceBuilder buildGetLatestLidVidsRequest(Collection<String> lids)
+    public static List<String> getAllLidVidsByLids(
+    		ControlContext ctlContext,
+    		RequestBuildContext reqContext,
+            Collection<String> lids) throws IOException
     {
-        if(lids == null || lids.isEmpty()) return null;
-        
-        SearchSourceBuilder src = new SearchSourceBuilder();
-        
-        // Query
-        BoolQueryBuilder boolQuery = QueryBuilders.boolQuery();
-        ProductQueryBuilderUtil.addArchiveStatusFilter(boolQuery);
-        boolQuery.must(QueryBuilders.termsQuery("lid", lids));        
-        src.query(boolQuery).fetchSource(false).size(0);
+    	List<String> lidvids = new ArrayList<String>();
 
-        // Aggregations
-        src.aggregation(AggregationBuilders.terms("lids").field("lid").size(lids.size())
-            .subAggregation(AggregationBuilders.topHits("latest").sort(new FieldSortBuilder("vid").order(SortOrder.DESC))
-                    .fetchSource(false).size(1))
-        );
-
-        return src;
+    	ctlContext.getConnection().getRestHighLevelClient().search(
+    			new SearchRequestFactory(RequestConstructionContextFactory.given("lid", new ArrayList<String>(lids)))
+    			.build(reqContext, ctlContext.getRegistryContext().getRegistryIndex()), RequestOptions.DEFAULT)
+    	.getHits().forEach((hit) -> { lidvids.add(hit.getId()); });
+    	return lidvids;
     }
-
     
-    /**
-     * Build terms query to select all document ids by a list of LIDs.
-     * @param lids a list of LIDS
-     * @return opensearch query
-     */
-    public static SearchSourceBuilder buildGetAllLidVidsRequest(Collection<String> lids)
+    public static String resolveLIDVID (
+    		String identifier,
+    		ProductVersionSelector scope,
+    		ControlContext ctlContext,
+    		RequestBuildContext reqContext) throws IOException, LidVidNotFoundException
     {
-        if(lids == null || lids.isEmpty()) return null;
-
-        BoolQueryBuilder boolQuery = QueryBuilders.boolQuery();
-        ProductQueryBuilderUtil.addArchiveStatusFilter(boolQuery);
-        boolQuery.must(QueryBuilders.termsQuery("lid", lids));        
-
-        SearchSourceBuilder src = new SearchSourceBuilder();
-        src.query(boolQuery).fetchSource(false).size(5000);
-        return src;
+    	String result = identifier;
+    	
+    	if (0 < identifier.length())
+    	{
+    		String lid = LidVidUtils.extractLidFromLidVid(identifier);
+    		/* YUCK! This should use polymorphism in ProductVersionSelector not a switch statement */
+    		switch (scope)
+    		{
```

Comment:
I think I see what it is about, is that how you implemented the /collections/{identifier} to return 404 if the id is not the one of a collection ? 

Somehow I don't like the way it is indented, but that is nitpicking.

Reply:
non sequitur : eclipse does a horrible job with indentation with me too. It looks good when I save it then when I load it on another day it is all over the place. Never sure if someone else edits it or just my eclipse. Maybe we all need to setup out eclipse to use spaces or tabs might help. I never changed mine from the factory default.

Not sure if you want changes or not. This is a uniform handling of lid/lidvid handler that everyone calls rather than having 10 different ways of doing the same. Like you bug below is fixed at this location only rather than having to do it 10 times at 11 different locations.